File: CSharp/Easy/LinkedList/RemoveNthNodeFromEnd.cs
/*
Example:

Given linked list: 1->2->3->4->5, and n = 2.

After removing the second node from the end, the linked list becomes 1->2->3->5.
*/
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        ListNode slow = head;
        ListNode fast = head;

       for(int i=0;i<n;i++)
       {
       		fast = fast.next;
       }
       if(fast==null)
       {
       		return head.next;
       }
       while(fast.next!=null)
       {
       		fast = fast.next;
       		slow = slow.next;
       }
       
       slow.next = slow.next.next;
       return head;
    }
}

File: Java/Easy/LinkedList/RemoveNthNodeFromEnd.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 1->2->3->4->5

 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
    	if(head==null)
    	{
    		return head;
    	}
        ListNode right = head;//1
        ListNode left = head;//1
        int i=0;
        while(i<n)//1
        {
			right = right.next;//null
			i++;//1
        }
        if(right==null)
        {
        	head = head.next;
        	return head;
        }
        while(right.next!=null)
        {
        	right = right.next;
        	left = left.next;
        }
        left.next = left.next.next;
        return head;
    }
}